hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/UpgradeDowngrade.java

```diff
@@ -95,6 +95,16 @@ public static boolean needsUpgrade(HoodieTableMetaClient metaClient, HoodieWrite
       throw new HoodieUpgradeDowngradeException(
           String.format("Please upgrade table from version %s to %s before upgrading to version %s.", fromTableVersion, HoodieTableVersion.SIX.versionCode(), toWriteVersion));
     }
+    // for table versions greater than or equal to six, if we are attempting to do an upgrade and auto-upgrade is disabled
+    if (fromTableVersion.greaterThanOrEquals(HoodieTableVersion.SIX) && fromTableVersion.versionCode() < config.getWriteVersion().versionCode() && !config.autoUpgrade()) {
```

Comment:
@yihua 
Had to add this check for table version greater than or equal to six otherwise this causes an issues for table versions 4 and 5 which we can not set the writeVersion to those values as write config value is only allowing 6,8,9

Reply:
will address this